src/object.lisp

```diff
@@ -174,17 +215,25 @@
 			       :name (cffi:foreign-funcall "g_type_name"
 							   :ulong gtype :string)))))))
 
+(defvar use-fake-objects nil)
```

Comment:
What is the purpose of this variable?

Reply:
Commit 95c96a242077764 ("Second Madhu patch. Fix for length-arg") introduced a code path that created "fake-objects" for types that are not exposed through g-i. e.g. what is returned by  (gir:invoke (*gio* "File" "new_for_path") "/etc/passwd"). - g_file_new_for_path is defined as returning an GFile which is GInterface, the actual type is some implementation dependent LocalFile type which is not available through g-i.

However I thought this code path may not be necessary , taking into account. Roman's commit 12f8910f205
if it were suitably extended. This patch implements the code path  which Roman probably intended 
to be the  default behaviour, while retaining the fake-object creation code path.
this variable provides a way to toggle the codepath -- I still think
the fake-object code path may be necessary in some situations but I don't know what they are
yet.

I tried explaining this briefly in the commit message.